basic_bot.py

```diff
@@ -0,0 +1,235 @@
+    max_val = np.fmax(val_1, val_2)
+    return np.fmin(max_val - min_val, min_val + size - max_val)
+
+
+def manhatten_distance(pos_1: int, pos_2: int, size: int) -> int:
+    coordinates_1 = get_coordinates(pos_1, size)
+    coordinates_2 = get_coordinates(pos_2, size)
+
+    dy = distance_1d(coordinates_1[0], coordinates_2[0], size)
+    dx = distance_1d(coordinates_1[1], coordinates_2[1], size)
+
+    return dx + dy
+
+
+# This function only has to be used once to create the look-up table in the beginning
+def create_distance_matrix(size: int) -> np.ndarray:
+    # Maximal position is size**2 - 1
+    position_range = np.arange(size ** 2)
+    positions_1 = np.repeat(position_range, size ** 2)
+    positions_2 = np.tile(position_range, size ** 2)
+
+    # positions_1 and positions_2 combined contain all possible combinations of positions
+    x_distances = distance_1d(positions_1 % size, positions_2 % size, size)
+    y_distances = distance_1d(positions_1 // size, positions_2 // size, size)
+
+    distance_matrix = (x_distances + y_distances).reshape(size ** 2, -1)
+
+    return distance_matrix
+
+
+# Score ########################################################################################################
+
+def score(board_halite: np.ndarray, board_shipyards: np.ndarray, ship: list,
+          player_id: int, size: int, distance_matrix: np.ndarray) -> np.ndarray:
+    halite_per_turn = np.zeros((size, size))
+
+    ship_position = ship[0]
+    ship_halite = ship[1]
+
+    shipyard_counts = np.column_stack(np.unique(board_shipyards[~np.isnan(board_shipyards)], return_counts=True))
+    shipyard_coordinates_array = np.column_stack(np.where(board_shipyards == player_id))
+
+    if shipyard_counts[shipyard_counts[:, 0] == player_id].size == 0:  # Player has no shipyard
+        halite_per_turn += 0.25 * board_halite / (1 + 2 * distance_matrix[ship_position].reshape(size, -1))
+
+    else:  # Player has at least one shipyard
+        distance_matrix_to_closest_shipyard = np.empty((size, size))
+        distance_matrix_to_closest_shipyard[:] = np.nan
+
+        for shipyard_coordinates in shipyard_coordinates_array:
+            shipyard_position = get_position(shipyard_coordinates, size)
+            distance_matrix_to_closest_shipyard = np.fmin(distance_matrix_to_closest_shipyard,
+                                                          distance_matrix[shipyard_position].reshape(size, -1))
+
+            halite_per_turn[shipyard_coordinates[0], shipyard_coordinates[1]] = \
+                drop_off_speed * ship_halite / (distance_matrix[ship_position, shipyard_position] + 1)
+
+        halite_per_turn += 0.25 * board_halite / (1 + distance_matrix[ship_position].reshape(size, -1)
+                                                  + distance_matrix_to_closest_shipyard)
+    return halite_per_turn
+
+
+def need_shipyard_(shipyards_count: int, ships_count: int) -> bool:
+    need_shipyard = False
+
+    if shipyards_count == 0:
```

Comment:
```suggestion
    if shipyards_count == 0 or ships_count / shipyards_count > 10:
          return True
    return False 
    
```

Löst das Problem in der Hälfte der Zeilen